applications/OptimizationApplication/custom_utilities/filtering/explicit_vertex_morphing_filter.h

```diff
@@ -0,0 +1,114 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   license: OptimizationApplication/license.txt
+//
+//  Main authors:    Baumgaertner Daniel
+//                   Aditya Ghantasala
+//                   Suneth Warnakulasuriya
+//
+
+#pragma once
+
+// System includes
+#include <string>
+#include <variant>
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "spatial_containers/spatial_containers.h"
+#include "expression/container_expression.h"
+
+// Application includes
+#include "entity_point.h"
+#include "filter_function.h"
+
+namespace Kratos {
+
+///@name Kratos Classes
+///@{
+
+template<class TContainerType>
+class KRATOS_API(OPTIMIZATION_APPLICATION) ExplicitVertexMorphingFilter
```

Comment:
Since this is going to be used in filtering any field it is not ExplicitVertexMorphingFilter, it should be named as simply ExplicitFiltering.

Reply:
I think ExplicitFiltering is too general as there are a lot of different methods. I will keep VM to explicetly define that VM methodology is used to compute A.

Reply:
@Igarizza As far as I know, based on an extensive literature study I did recently, convolution-based filtering is coined as explicit filtering and it has been used in node-based shape optimization (VM) as well as topology and other field optimizations. As described in the description, this kind of filtering is container agnostic, so can be used for any nodal, elemental, and conditional field.